database_interactions/mongodb/src/main.rs

```diff
@@ -0,0 +1,47 @@
+use actix_web::{App, HttpServer};
+use dotenv::dotenv;
+use mongodb::{options::ClientOptions, Client};
+use service::UserService;
+use std::sync::Arc;
+
+mod handlers;
+mod service;
+
+struct MongoDB {
+    client: Client,
+}
+impl MongoDB {
+    async fn init() -> Result<Self, mongodb::error::Error> {
+        let client_uri = std::env::var("MONGODB_URI").expect("set MONGODB_URI env var");
+        let client_options = ClientOptions::parse(&client_uri).await?;
+        Ok(MongoDB {
+            client: Client::with_options(client_options)?,
+        })
+    }
+    fn connection_base(&self, base: &str) -> mongodb::Database {
+        self.client.database(base)
+    }
+}
+
+struct AppState {
+    user: UserService,
+}
+
+#[actix_web::main]
+async fn main() -> std::io::Result<()> {
+    dotenv().ok();
+    let mongo = MongoDB::init().await.expect("mongodb connection");
+    let db = mongo.connection_base("myApp");
+    let app_state = Arc::new(AppState {
+        user: UserService::new(db.collection("users")),
+    });
```

Comment:
Maybe is better to use `web::Data` here instead of `Arc` and `app_data` instead of `data` inside `HttpServer::new`, since Data already uses an `Arc`.

Reply:
In this case type it would be  `web::Data<Mutex<my_data>>` ?
 I'm starting my trail with rust, thanks for help.

Reply:
Thank you for the PR. I think you just need to change `Arc::new` to `web::Data::new` and use the `app_data` method instead of the `data` in the `App` type. Let me know if you find any problems.